### PR TITLE
Add negative authorization test to E2E script

### PR DESCRIPTION
**Pull request by okhasawn**
### Description
This PR adds a test to the E2E script that sends unauthorized credentials to the clusters.
This ensures the security of the authentication process and validates that unauthorized access is prevented.

### Issues Resolved
[MIGRATIONS-1153](https://opensearch.atlassian.net/browse/MIGRATIONS-1153)

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/tests.py [hunk 1]
```diff
@@ -160,3 +162,20 @@ def test_003_jupyterAwake(self):
         # Making sure that the Jupyter notebook is up and can be reached.
         response = requests.get(self.jupyter_endpoint)
         self.assertEqual(response.status_code, HTTPStatus.OK)
+
+    def test_004_negativeAuth(self):
```
Comment (sumobrian) [inline test/tests.py:166]: I would add what the test is doing to the function name. For example,
test_004_negativeAuth_InvalidCreds.

Comment (sumobrian) [reply test/tests.py:166]: Also I would add at least one more 0 to your test case numbers.

**Review by sumobrian — CHANGES_REQUESTED** [review]
I mentioned that you should break up test cases into separate review but I think there are few more related test cases you should add.

1. Missing Authentication: Attempt to access resources without any authentication details.

2. Invalid Authentication: Use expired, malformed, or otherwise invalid tokens, session IDs, or credentials.

I would normally add authorization tests too (where you are testing insufficient privileges) but in this case, save these for another PR.

**Comment by okhasawn** [comment]
> I mentioned that you should break up test cases into separate review but I think there are few more related test cases you should add.
> 
>     1. Missing Authentication: Attempt to access resources without any authentication details.

Just addressed in a different test
 
>     2. Invalid Authentication: Use expired, malformed, or otherwise invalid tokens, session IDs, or credentials.
> 
> 

The cluster we currently deploy doesn't use token-based authentication, so at this point, this is not possible. BUT, just took a quick look and found it might be possible to use both basic and token-based authentication.

**Review by sumobrian — APPROVED** [review]

**Pull request merged**